ballerina/schema_types.bal

```diff
@@ -91,10 +91,15 @@ public type EdiFieldSchema record {|
     boolean truncatable = true;
     EdiDataType dataType = STRING;
     int startIndex = -1;
-    int length = -1;
+    Range|int length = -1;
     EdiComponentSchema[] components = [];
 |};
 
+public type Range record {|
+    int min = 0;
+    int max = -1;
```

Comment:
Can max be -1?

Reply:
In here -1 will treat as a infinite or unset max length. Is there any other standard way of doing that?

Reply:
Can't we make it optional? What is the standard in other specifications?

Reply:
This `schema_types` use to make a `cloneWithType()` of schema given by the **user**. Therefore in **user** perspective the fields are **optional** for their schemas. Since we use generated normalized schema as a intermediate data to parsers it is easy to us to make it like this.

Reply:
Lets document these for Range record and proceed

Reply:
Sure